pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -176,52 +191,60 @@ func (f *FederatedTypeConfig) GetObjectMeta() metav1.ObjectMeta {
 	return f.ObjectMeta
 }
 
-func (f *FederatedTypeConfig) GetTarget() metav1.APIResource {
-	return apiResourceToMeta(f.Spec.Target, f.Spec.Namespaced)
+func (f *FederatedTypeConfig) GetTargetType() metav1.APIResource {
+	return apiResourceToMeta(f.Spec.TargetType, f.GetNamespaced())
 }
 
+// TODO(font): This method should be removed from the interface in favor of
+// checking the namespaced property of the appropriate APIResource (TargetType,
+// FederatedType) depending on context.
 func (f *FederatedTypeConfig) GetNamespaced() bool {
-	return f.Spec.Namespaced
+	return f.Spec.TargetType.Namespaced()
 }
 
 func (f *FederatedTypeConfig) GetPropagationEnabled() bool {
-	return f.Spec.PropagationEnabled
+	return f.Spec.Propagation == PropagationEnabled
 }
 
 func (f *FederatedTypeConfig) GetFederatedType() metav1.APIResource {
 	return apiResourceToMeta(f.Spec.FederatedType, f.GetFederatedNamespaced())
 }
 
-func (f *FederatedTypeConfig) GetStatus() *metav1.APIResource {
-	if f.Spec.Status == nil {
+func (f *FederatedTypeConfig) GetStatusType() *metav1.APIResource {
+	if f.Spec.StatusType == nil {
 		return nil
 	}
-	metaAPIResource := apiResourceToMeta(*f.Spec.Status, f.Spec.Namespaced)
+	metaAPIResource := apiResourceToMeta(*f.Spec.StatusType, f.Spec.StatusType.Namespaced())
 	return &metaAPIResource
 }
 
-func (f *FederatedTypeConfig) GetEnableStatus() bool {
-	return f.Spec.EnableStatus
+func (f *FederatedTypeConfig) GetStatusEnabled() bool {
+	return f.Spec.StatusCollection != nil && *f.Spec.StatusCollection == StatusCollectionEnabled
 }
 
-// TODO(marun) Remove in favor of using 'true' for namespaces and the
-// value from target otherwise.
+// TODO(font): This method should be removed from the interface i.e. remove
+// special-case handling for namespaces, in favor of checking the namespaced
+// property of the appropriate APIResource (TargetType, FederatedType)
+// depending on context.
 func (f *FederatedTypeConfig) GetFederatedNamespaced() bool {
 	// Special-case the scope of federated namespace since it will
 	// hopefully be the only instance of the scope of a federated
 	// type differing from the scope of its target.
 
-	// TODO(marun) Use the constant in pkg/controller/util
 	if f.IsNamespace() {
-		// Namespace placement is namespaced to allow the control
-		// plane to run with only namespace-scoped permissions.
+		// FederatedNamespace is namespaced to allow the control plane to run
+		// with only namespace-scoped permissions to determine placement.
```

Comment:
(No action required) I think the addition of 'to determine placement' is not helpful because that's not the only reason for the discrepancy.  Since this function will be shortly be removed, though, I won't ask for any change.

Reply:
Updated to reflect that it's just an example.